src/tooltip/Tooltip.tsx

```diff
@@ -1,357 +1,224 @@
       >
-        {this.renderContent(false)}
-        <ModalComponent
-          animationType="fade"
-          visible={isVisible}
-          transparent
-          onShow={onOpen}
+        <Pressable
+          {...(!closeOnlyOnBackdropPress && { [toggleAction]: toggleToolTip })}
         >
-          {this.renderModalContent()}
-        </ModalComponent>
+          {children}
+        </Pressable>
+        <Modal transparent visible={visible}>
+          <TouchableWithoutFeedback onPress={toggleToolTip}>
+            <Animated.View
+              style={[
+                StyleSheet.absoluteFillObject,
+                {
+                  backgroundColor: overlayColor || '#f1f1f188',
+                  opacity: animation,
+                },
+              ]}
+              pointerEvents={visible ? 'auto' : 'none'}
+            />
+          </TouchableWithoutFeedback>
+          <SafeAreaView>
+            <Animated.View
+              pointerEvents={visible ? 'auto' : 'none'}
+              style={[
+                styles.tooltip,
+                {
+                  opacity: animation.interpolate({
+                    inputRange: [0, 0.4, 0.8, 1, 1],
+                    outputRange: [0, 0, 0, 1, 1],
+                  }),
+                },
+              ]}
+            >
+              {withPointer ? (
+                <Triangle
+                  isDown={ScreenHeight / 2 < containerDimensions.py}
+                  style={{
+                    position: 'absolute',
+                    borderBottomColor: pointerColor || backgroundColor,
+                    top:
+                      containerDimensions.py -
+                      containerDimensions.height *
+                        (ScreenHeight / 2 < containerDimensions.py ? 2 : 0) -
+                      12,
+                    left:
+                      containerDimensions.px + containerDimensions.width / 2,
+                  }}
+                />
+              ) : null}
+              <View
+                style={[
+                  styles.tooltipView,
+                  {
+                    maxWidth:
+                      ScreenWidth / 2 > propWidth ? propWidth : ScreenWidth / 3,
+                    backgroundColor,
+                  },
+                  ToolTipStyle(),
+                  containerStyle,
+                ]}
+                ref={tooltipElement}
+                onLayout={(e) => Object(e.nativeEvent.layout)}
+              >
+                {title ? <Text style={styles.label}>{title}</Text> : popover}
+              </View>
+            </Animated.View>
+          </SafeAreaView>
+        </Modal>
       </View>
-    );
-  }
-}
+    </>
+  );
+};
+
+const styles = StyleSheet.create({
+  content: {
+    position: 'relative',
+  },
+  tooltip: {
+    position: 'absolute',
+    top: 10,
+    left: -10,
+  },
+  label: {
+    color: '#fff',
+  },
+  tooltipView: {
+    padding: 10,
+    maxWidth: 200,
+    width: ScreenWidth / 2,
```

Comment:
What about landscape mode ?

Reply:
![Screenshot_1623415930](https://user-images.githubusercontent.com/55053424/121689219-fca42400-cae1-11eb-8bc3-94265ed759f3.png)
![Screenshot_1623415934](https://user-images.githubusercontent.com/55053424/121689214-fada6080-cae1-11eb-971b-603820d15154.png)

Reply:
There is issue with Modal on web, Can we make `tooltip.web.tsx` and do without it?

Reply:
No, because then users have to change the configs here and there. `.native.js` `.web.js`

Reply:
https://github.com/expo/expo/issues/7034

Reply:
What happens when you switch back to portrait w/o closing the app, does it still work fine?

Reply:
Not now, I will add a listener

Reply:
@renjfk 
![rotate](https://user-images.githubusercontent.com/55053424/121736474-3d696080-cb15-11eb-954d-62c1279b824c.gif)